CMakeLists.txt

```diff
@@ -0,0 +1,150 @@
+# Forbid CMake from editing sources
+set(CMAKE_DISABLE_SOURCE_CHANGES ON)  # NB: currently undocumented.
+
+# By default, do not output "Up-to-date" messages during install. Can
+# be spammy when this project is included in another via CMake's
+# ExternalProject.
+if (NOT DEFINED CMAKE_INSTALL_MESSAGE)
+    set(CMAKE_INSTALL_MESSAGE LAZY)
+endif ()
+
+
+#-----------------------------------------------------------------------
+# Project metadata
+
+string(CONCAT project_description
+    "An open-source interoperability standard for tools and content"
+    " management systems used in media production.")
+
+project(
+    OpenAssetIO
+    VERSION 0.0.0
+    DESCRIPTION ${project_description}
+    HOMEPAGE_URL https://thefoundryvisionmongers.github.io/OpenAssetIO
+    LANGUAGES C CXX)
+
+
+#-----------------------------------------------------------------------
+# Global settings
+
+include(ProjectSettings)
+
+
+#-----------------------------------------------------------------------
+# Options
+
+# Installation options
+# TODO(DF): Install dir wrangling depending on desired Python/C++
+#   library combination. See OTIO.
+
+# Build options
+# TODO(DF): This means CMake controls whether static or shared libs are
+#   created with `add_library` when no lib type is specified. Perhaps
+#   we shouldn't use this built-in CMake variable, since it can affect
+#   dependencies too (e.g.
+#   https://github.com/boostorg/boost_install/issues/5). See also OTIO.
+option(BUILD_SHARED_LIBS "Set to OFF to build static libraries" ON)
+
+
+#-----------------------------------------------------------------------
+# Enable ctest testing
+
+# To correctly generate the DartConfiguration.tcl, include the CTest
+# CMake file. By doing so 'ctest -F memcheck' works.
+# TODO(DF): ...but also creates many many pointless targets.
+#include(CTest)  # Automatically calls enable_testing()
+enable_testing()
+
+#-----------------------------------------------------------------------
+# Provides install directory variables as defined by the GNU Coding
+# Standards.
+
+include(GNUInstallDirs)
+
+#-----------------------------------------------------------------------
+# Generate OpenAssetIOConfig.cmake et al. for bundling as a CMake lib.
+
+# Targets exported to ${PROJECT_NAME}_EXPORTED_TARGETS will be available
+# for import using find_package(OpenAssetIO).
+
+include(CMakePackageConfigHelpers)
+
+set(version_config_file "${CMAKE_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake")
+set(project_config_file "${CMAKE_BINARY_DIR}/${PROJECT_NAME}Config.cmake")
+set(config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+
+# Version fetched from the top level project()
+write_basic_package_version_file(
+    ${version_config_file}
+    # Version range supported only if generated by CMake >= 3.19.2.
+    COMPATIBILITY SameMinorVersion
+)
+
+configure_package_config_file(
+    cmake/Config.cmake.in ${project_config_file}
```

Comment:
Nit: personal preference is to put different arguments in CMake calls on separate lines - just makes it easier to read for me.

Exception is when there are key-value pairs, e.g. keyword + value, in which case I normally put them on a single line if it makes sense to.

Reply:
I like the sound of that. It's good to have some guidance on code style so please nitpick away!

Reply:
Conformed the style in 14fa4b78